bin/common.bash

```diff
@@ -333,3 +344,95 @@ ck8s_kubespray_version_check(){
+    # Check for AMS
+    primary_group_label=$(yq4 .group_label_primary "${config_path}/group_vars/all/ck8s-kubespray-general.yaml")
+    secondary_group_label=$(yq4 .group_label_secondary "${config_path}/group_vars/all/ck8s-kubespray-general.yaml")
+    if [[ $(ops_kubectl "$prefix" get node "$node" -ojson | jq ".metadata.labels | has(\"${control_plane_label}\")") == "true" ]]; then
+        target_group="kube_control_plane"
+        if [[ "$(group_exists "${config[groups_inventory_file]}" "$target_group")" != "true" ]]; then
+            log_info "Adding $target_group group to ${config[groups_inventory_file]} .."
+            add_group "${config[groups_inventory_file]}" "$target_group"
+        fi
+        add_host_to_group "${config[groups_inventory_file]}" "$node" "$target_group"
+    elif [[ $(ops_kubectl "$prefix" get node "$node" -ojson | jq ".metadata.labels | has(\"${primary_group_label}\")") == "true" ]]; then
+        node_type=$(ops_kubectl "$prefix" get node "$node" -ojson | jq -r ".metadata.labels[\"${primary_group_label}\"]")
+        if [[ $(ops_kubectl "$prefix" get node "$node" -ojson | jq ".metadata.labels | has(\"${secondary_group_label}\")") == "true" ]]; then
+            cluster_name=$(ops_kubectl "$prefix" get node "$node" -ojson | jq -r ".metadata.labels[\"${secondary_group_label}\"]")
+            target_group="${node_type}_${cluster_name//-/_}"
+        else
+            target_group="${node_type}"
+        fi
+        if [[ "$(group_exists "${config[groups_inventory_file]}" "$target_group")" != "true" ]]; then
+            log_info "Adding $target_group group to ${config[groups_inventory_file]} .."
+            add_group "${config[groups_inventory_file]}" "$target_group"
+        fi
+
+        if [[ "$node_type" == "postgres" ]]; then
+            if [[ $(ops_kubectl "$prefix" get pods -A -l application=spilo,cluster-name="$cluster_name" -ojson | jq -r '.items[] | select( .metadata.labels["spilo-role"] == "master" ).spec.nodeName') == "$node" ]]; then
+                add_host_to_group_as_last "${config[groups_inventory_file]}" "$node" "$target_group"
+            else
+                add_host_to_group "${config[groups_inventory_file]}" "$node" "$target_group"
+            fi
+        else
+            add_host_to_group "${config[groups_inventory_file]}" "$node" "$target_group"
+        fi
+
+
+    # Check for regular nodes
+    else
+        target_group="regular_worker"
+        if [[ "$(group_exists "${config[groups_inventory_file]}" "$target_group")" != "true" ]]; then
+            log_info "Adding $target_group group to ${config[groups_inventory_file]} .."
+            add_group "${config[groups_inventory_file]}" "$target_group"
+        fi
+        add_host_to_group "${config[groups_inventory_file]}" "$node" "$target_group"
+    fi
+}
+
+contains_element () {
+  local e match="$1"
+  shift
+  for e; do [[ "$e" == "$match" ]] && return 0; done
```

Comment:
How does this work exactly? `e` doesn't seem to be assigned anywhere. And still you can iterate over it and use the same name as the element? 
It would be nice, imo, if you could expand this a little bit just to make it a bit more readable

Reply:
Yep, I agree, that might be confusing, as the goal is to check if an element exist in an array.

**shift** shifts the argument list by 1 to the left (dropping the first argument) and for without an in implicitly iterates over the argument list

Reply:
Right, it was that `for` magic with out using `in` that I didn't know about. But maybe just expand it to

```bash
for e in "$@"; do
  [[ "$e" == "$match" ]] && return 0
done
```

Would make it way more readable in my book.

Reply:
Sure, will change that. Thanks a lot